vpr/src/route/check_route.cpp

```diff
@@ -463,14 +478,34 @@ static bool check_adjacent(RRNodeId from_node, RRNodeId to_node, bool is_flat) {
                     /* UDSD Modification by WMF End */
                 }
             } else if (to_type == e_rr_type::CHANX) {
-                num_adj += chanx_chany_adjacent(to_node, from_node);
+                num_adj += rr_graph.nodes_are_adjacent(to_node, from_node);
+            } else if (to_type == e_rr_type::CHANZ) {
+                num_adj += chanxy_chanz_adjacent(from_node, to_node);
```

Comment:
Let's push this code into .nodes_are_adjacent. Can have it call the current helper after figuring out there is a CHANZ and swapping nodes (or do the swapping at the top of the current routine).